atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/watch/ResilientLockWatchEventCache.java

```diff
@@ -34,6 +34,15 @@
 final class ResilientLockWatchEventCache extends AbstractInvocationHandler {
 
     private static final Logger log = LoggerFactory.getLogger(ResilientLockWatchEventCache.class);
+    private static final Method IS_ENABLED;
+
+    static {
+        try {
+            IS_ENABLED = LockWatchEventCache.class.getMethod("isEnabled");
+        } catch (NoSuchMethodException e) {
+            throw new RuntimeException("Did someone rename this method?", e);
+        }
+    }
```

Comment:
I don't really like this - I much prefer just implementing in LWECI and NOLWEC as I feel it is less error prone, and doing this kind of proxy-wrangling is abusing the whole point of it IMO.

Reply:
Discussed offline